internal/controller/resourcesetinputprovider_controller.go

```diff
@@ -169,15 +180,37 @@ func (r *ResourceSetInputProviderReconciler) reconcile(ctx context.Context,
 func (r *ResourceSetInputProviderReconciler) newGitProvider(ctx context.Context,
 	obj *fluxcdv1.ResourceSetInputProvider,
 	certPool *x509.CertPool,
-	username, password string) (gitprovider.Interface, error) {
+	username, password string,
+	authData map[string][]byte) (gitprovider.Interface, error) {
 	switch {
 	case strings.HasPrefix(obj.Spec.Type, "GitHub"):
+		var ghAppToken string
+		if username == "" {
+			var err error
+			ghAppToken, err = r.getGitHubAppToken(ctx, authData)
+			if err != nil {
+				return nil, fmt.Errorf("failed to get GitHub app token: %w", err)
+			}
+		}
+		if authData != nil && username == "" && ghAppToken == "" {
+			return nil, fmt.Errorf("invalid auth configuration: either basic auth or GitHub app auth is required")
+		}
+		if username != "" && ghAppToken != "" {
+			return nil, fmt.Errorf("invalid auth configuration: both basic auth and GitHub app auth are provided")
+		}
```

Comment:
This is dead code due to the logic above where `getGitHubAppToken` gets called only when username is empty.

Reply:
great catch!

Reply:
fixed, thanks